File: 010/010.py
import math

limit = 2000000
sievebound = (limit - 1) // 2
crosslimit = int((math.sqrt(limit) - 1) / 2)
sieve = [False for i in range(2, sievebound+2)]

for i in range(1, crosslimit):
    if not sieve[i]:
        # 2*i+1 is prime, mark mulitples
        for j in range(2*i*(i+1), sievebound, 2*i+1):
            sieve[j] = True

s = 2
for i in range(1, sievebound):
    if not sieve[i]:
        s += 2*i+1

print(s)

File: 010/010.cc
#include <iostream>
#include <cmath>

int main() {
    int limit = 2000000;
    int sievebound = (limit - 1) / 2;
    int crosslimit = (int) (sqrt(limit) - 1) / 2;
    int *sieve = new int[sievebound];

    for (int i = 1; i < crosslimit; i++)
        if (sieve[i] == 0)
            for (int j = 2*i*(i+1); j < sievebound; j += 2*i+1)
                sieve[j] = 1;

    long sum = 2;
    for (int i = 1; i < sievebound; i++)
        if (sieve[i] == 0)
            sum += 2*i+1;

    std::cout << sum << std::endl;
}
